### ISSUE TITLE
[SPELL DELAY RESEARCH]

### ISSUE THREAD

**Issue by Evolvee** [Evolvee/Spell-Delay-Research#1]
### NOTE THAT THINGS LISTED HERE ARE JUST A SMALL % OF WHAT I ACTUALLY RESEARCHED, I JUST MOSTLY FORGOT TO NEED TO EVEN MENTION IT SINCE AFTER ALL THESE YEARS IM TAKING IT AS GRANTED ALREADY (open, sealed batches etc.)…
**This post does in no way include all researched mechanics or the entire evidence/material for further research.**

-----------------

**DISCLAIMER:** ALL NETHERWING VIDEOS IN THIS POST WERE RECORDED DURING SUMMER 2017 WHEN WE FOCUSED ON SPELL DELAY --> before it got cucked by Wolfenstein due to his changes to core due to which he could no longer restore the initial version.
(tldr: the current version, broken mess, is no longer the same as it was)

**DISCLAIMER 2**: Despite the army of devs and multimillionaire budget, Blizzard is extremely retarded and the entire game is a piece of shit from common sense and logic-wise when it comes to mechanics and their behavior.

-----------------

Its worth to mention that the most effective classes for spell delay research were:
1) shadow priest (I wonder why 😉 )
2) mage
3) rogue

The rest is mostly just confirming the mechanics shown in classes above or have some interesting exceptions to them.

This post is in no means finished and will get eventually updated if I continue my research some day.

Credits:
[Evolve](https://youtu.be/xAdo9bgToTk) (research part, bug-reporting, quality control afterwards)
[Wolfenstein](https://youtu.be/-H7jQ9arR_A) (Implementation on Netherwing)

----------------

Btw, I dont know what random came up with it recently, but it is called nothing but **SPELL DELAY**! Not batching. Not spell batching. Not batch system. It is SPELL DELAY.  (derived from Corecraft)
_Thanks for understanding._

**Comment by Evolvee** [comment]
**Grounding Totem**
https://youtu.be/RbeqSGY0tM8?t=419 --> absorbs both fb and elementals fb that fired AFTER fb fired

https://github.com/ccshiro/cc-buglist/issues/1682#issuecomment-141958648 --> grounding totem should absorb even procs from absorbed spells 

http://eu.battle.net/wow/en/game/patch-notes/2-0-1 -->
> Fixed an issue where a spell with two procs would be reflected and one of the procs would still hit the player that reflected the spell.

^^ even two procs = 400ms + 400ms




---------------

**Vanish FADE OUT TIME DELAY**

=> This delay is tied to a mechanic called an _"OBJECT DELAY"_

On retail you could see Vanish buff in targets PlayerFrame buff bar before he actually disappeared --> that was the time frame where you could autoattack(we disabled that on NW) or spell cast into it and break the vanish... (note that the ability that breaks the vanish should most likely be applied on top of it as well - however, sometimes it isnt - dunno why)
This is called fade-out time delay.
I believe its way shorter duration than 400ms otherwise ppl would break vanish all the time = it must be another batch that doesnt perfectly sync with object-visibility delay...

https://www.youtube.com/watch?v=xF4Rx1sQYQU&feature=youtu.be&t=91

![image](https://user-images.githubusercontent.com/1942409/47362868-b9e57180-d6d5-11e8-9bc9-342f15f37337.png)


^^here you can see the vanish and stealth buffs before the player disappears
Note, that GCD golden delay is not synced with this object/fade out vanish delay. You can see blind golden delay being executed and the player would disappear in like another 200ms, its not synced again..

This was the order:
1. rogue trinketed
2. enemy rogue casted vanish, instantly got vanish buff and stealth buff on himself = fade out time delay
3. rogue in POV casted blind, got golden GCD delay
4. golden GCD delay ended, enemy rogue still didnt disappear
5. enemy rogue disappears
6. vanish is canceled because rogue from POV casted the blind into the vanish while it was in fade out time delay

You can see literally the same scenario in clips that follow this timestamped one...

Another example here: https://www.youtube.com/watch?v=GYUbaSCjCTU&feature=youtu.be&t=40
You see rogue vanishing and druid feral charing into it, removing the vanish and applying feral charge.

![image](https://user-images.githubusercontent.com/1942409/47363359-e1890980-d6d6-11e8-93f6-03185d1353b9.png)


rogue had fade out time vanish delay on himself before feral charge was queued
Note, that feral charge was executed instantly right after it was queued - there was no GCD golden delay.

![image](https://user-images.githubusercontent.com/1942409/47363404-fd8cab00-d6d6-11e8-9a86-e38442c8a089.png)

You can see the rogue actually disappeared for a brief moment before getting "immune" message from druids feral charge and then appeared again due to vanish break by that feral charge.


https://github.com/ccshiro/cc-buglist/issues/1122#issuecomment-191648653

> Personally, I suspect that this has to do with the small amount of time where you can see the vanish buff being applied to your opponent and him disappearing. When I last played on the test server, this time window seemed unsually large (on retail, when I played as and against rogues, it was usually much shorter - anecdotal).
> 
> It might be, that auto attacks during this window (and spell casts) aren't stopped as they're supposed to be. It might be that all attacks should stop instantly, before the vanishing rogue even disappears from sight.
> 
> I specifcally remember when 3.0.2 came out and people played it, how important vanish not breaking was for mage/rogue. I remember watching them play and playing against them and having guildies complain, if it happened. But it definitely wasn't as frequent as "80% of vanishes in melee range". That's including the /stopattack for your own autohits of course. Not that I see the point of purposely recreating bugs if we're not having seal twisting anyway.

![image](https://user-images.githubusercontent.com/1942409/47363648-96232b00-d6d7-11e8-8bfc-4beb2d5bad50.png)


https://youtu.be/eIc9UzUCpF8?t=687

Here you can see it very clearly - vanish gets deleted by vampiric embrace and the embrace gets applied through vanish in delay,essentially removing both vanish and stealth. And gaining **COMBAT** at the exact same batch synced with the destealth and aura application batch:
![image](https://user-images.githubusercontent.com/1942409/47363758-dedae400-d6d7-11e8-91bc-3e6c99826761.png)

Notable examples:
https://youtu.be/RbeqSGY0tM8?t=206 --> U can clearly see both vanish AND stealth before he disappears

https://youtu.be/i0uFJeZOGuo?t=92 --> here you can see getting kidney shotted into vanish

https://youtu.be/Ri4otugfny0?t=464 --> this is old vanilla video, but here you can see Vurtne canceling vanish with counterspell

https://youtu.be/Ri4otugfny0?t=818 --> Again Vurtne canceling vanish with counterspell

https://youtu.be/WJTRmRIUvVA?t=371--> canceled vanish with fire blast

https://www.youtube.com/watch?v=U7wwsXO2Njc&feature=youtu.be&t=619 --> canceled vanish by i dont even know what

From warmane bugtracker:

(note that this is a report from ppl who didnt do the research as I did so they dont know about many things, so take it as a little bonus)
![image](https://user-images.githubusercontent.com/1942409/27458253-666c086e-57a8-11e7-9f23-d1f259c994bd.png)
![image](https://user-images.githubusercontent.com/1942409/27458270-72a78b9e-57a8-11e7-8393-424d4116cfea.png)
![image](https://user-images.githubusercontent.com/1942409/27458282-829bca7e-57a8-11e7-8f40-23d593f21da6.png)


-------------

**Comment by Evolvee** [comment]
**VANISH 2** (vanishing instant and projectile spells)

when you vanish flying spell(projectile), you SHOULD "dodge" that projectile and still be in vanish

but when you vanish instant spell(earth shock/arcane explosion/cone of cold), you SHOULD "dodge" that spell but IT SHOULD BREAK YOUR VANISH/STEALTH actually

NOTE: All instants **EXCEPT shadowfury!** 
NOTE 2:  There seems to be another exception, that is more detailed down below.ˇˇ

https://www.youtube.com/watch?v=fT_aZfg9AP0&feature=youtu.be&t=41s
vanished death coil(projectile spell) --> still in vanish

https://www.youtube.com/watch?v=V0UvkL-mZq8&feature=youtu.be&t=13m42s
vanished death coil(projectile spell) --> still in vanish

https://youtu.be/xF4Rx1sQYQU?t=62
vanished blind(instant spell) --> vanish+stealth broken

https://youtu.be/xF4Rx1sQYQU?t=74
vanished blind(instant spell) --> vanish + stealth broken

https://github.com/ccshiro/cc-buglist/issues/1122

https://youtu.be/TTDbFBbNgyw?t=97
instant spells break vanish

https://youtu.be/TTDbFBbNgyw?t=110
death coil(projectile) leaves you in stealth

https://youtu.be/TTDbFBbNgyw?t=130
frostbolt, polymorph etc.

https://github.com/ccshiro/cc-buglist/issues/1122#issuecomment-129041530

> Also, other rogues can vanish my cheap shot, I will get out of stealth, but their vanish will not break.
> This is not how it is supposed to work. If you vanish an instant spell that isn't a projectile (like blind, cheap shot, etc) you are immune to the spell, but vanish/stealth also breaks.


https://www.youtube.com/watch?v=fT_aZfg9AP0&t=663
cheap shot vs vanish = vanish break

https://github.com/ccshiro/cc-buglist/issues/1122#issuecomment-131403291

> Instant spells break vanish all the time when they get avoided by it. Maybe projectiles break it sometimes. The so-called vanish bug was common knowledge on retail. However, I'd be very surprised if you could find a video of an instant spell (no projectile) being swallowed by vanish without it breaking.


https://github.com/ccshiro/cc-buglist/issues/1122#issuecomment-131650123
https://github.com/ccshiro/cc-buglist/issues/1122#issuecomment-136000173
https://github.com/ccshiro/cc-buglist/issues/1122#issuecomment-136027009
https://github.com/ccshiro/cc-buglist/issues/1122#issuecomment-136037666
https://github.com/ccshiro/cc-buglist/issues/1122#issuecomment-174127274

> actually, if someone still intrested, DT doesn't breakes Vanish just like all other spells with travel time. But the rogue's PvP gloves bonus does, since it gives you instant interrupt.

https://github.com/ccshiro/cc-buglist/issues/1122#issuecomment-183772683

https://www.youtube.com/watch?v=aZlkojMaJ3w&feature=youtu.be&t=1063
noone frostbolt---> vanish didnt break because FB is flying spell

http://www.arenajunkies.com/topic/64325-rogue-how-to-vanish-serious-question/

> Vanishing projectile spells through reaction:
> You wait till the cast flies towards you and then you vanish, the spell is absorbed and you are still in stealth.
> 
> Vanishing non-projectile spells through prediction:
> There is this movie starring a rogue (I think it’s Akrios) in which he predicts the hammer of justice, and successfully vanishes it (read: vanish, HoJ, vanish breaks – no HoJ either).


https://youtu.be/Ba6Ty72eSH8?t=261

vanished floating ice lance, stealth is not removed(because its projectile, not instant spell)

https://youtu.be/ewlXQcYEKSg?t=10

https://www.youtube.com/watch?v=E8hQG5CnG5s

**NOTE! 1 exception probably:**
https://www.warmane.com/bugtracker/report/45434
![image](https://user-images.githubusercontent.com/1942409/47386905-3300bb00-d70e-11e8-9ca8-d121640216fd.png)

https://www.warmane.com/bugtracker/report/45435
![image](https://user-images.githubusercontent.com/1942409/47387007-6f341b80-d70e-11e8-9f24-710de22b8e14.png)


**OFFTOPIC**: Pre-cast + VANISH AND COMBAT:
https://www.warmane.com/bugtracker/report/48970
![image](https://user-images.githubusercontent.com/1942409/47387055-91c63480-d70e-11e8-895f-0d41db3a597f.png)


https://youtu.be/WQEnZKOZxtg?t=95 Blind vanish sap
https://youtu.be/RzMmtKyHbvA?t=264 FoK vanish sap
https://youtu.be/RzMmtKyHbvA?t=161 Ambush vanish sap
https://youtu.be/RzMmtKyHbvA?t=178 Eviscerate and lightning trinket vanish sap

^^ I didnt research this, so take it as an offtopic.

**Comment by Evolvee** [comment]
**Vanish 3 - Delayed unstealth mechanics, object delay**
(Also known as _DELAYED VISIBILITY SYSTEM_)

Currently(without this mechanic in place with "fake delay in place") NPCs are able to hit you before your cheapshot is applied...
Same will go for players most likely so ppl will most likely be able to blind you if you open them with cheapshot before the stun gets actually applied.


https://youtu.be/xF4Rx1sQYQU?t=72
this might be seen here: cheashot + unstealth delay

we can clearly see that he isnt unstealthed until he get combat(cheapshot application)

https://youtu.be/xF4Rx1sQYQU?t=72
Note that he got combat the exact same moment his bar stance switched.
So stealth bar switch is bound on combat.

https://www.youtube.com/watch?v=xF4Rx1sQYQU&feature=youtu.be&t=91
Same as ^^ combat synced with bar stance switch.

In Acrono video you can see that the stealth buff and cheapshot debuff are synced. Thats caused by the aura application delay, also known as aura/debuff batch delay. --> we will get to it later

https://youtu.be/Ba6Ty72eSH8?t=294
https://youtu.be/Ba6Ty72eSH8?t=332
https://youtu.be/Ba6Ty72eSH8?t=350
https://youtu.be/Ba6Ty72eSH8?t=435

https://youtu.be/Ba6Ty72eSH8?t=494
rogue vs rogue, as u can see garrote is applied at the exact same mooment as rogue is visible to the other rogue

https://youtu.be/Ba6Ty72eSH8?t=529
garrote same thing
https://youtu.be/Ba6Ty72eSH8?t=543

also note that the mage is getting the target of target of Acrono after a while(after cheapshot + some delay)

mage getting cheapshoted pov:
https://youtu.be/K4WR0JW5jWM?t=694
noone didnt see the rogue before cheapshot proced(and for a little while while he was already cheapshoted)

Schaka skype:
[17:48:52] Schaka Garcia: I don't think visual update happened before cheap shot was applied
[17:49:02] Schaka Garcia: so you couldn't react and blind before cs hits you

Notable examples:
https://youtu.be/Ba6Ty72eSH8?t=570
https://youtu.be/Ba6Ty72eSH8?t=609
https://youtu.be/Ba6Ty72eSH8?t=708
^^
stealth removal + cheap is synced perfectly which prevents the blind exploit

Its crucial to understand that in ALL these clips, its either 100% synced, or in 2nd players PoV on which the CC is casted from stealth, its perfectly visible that HE CANNOT SEE THE ROGUE UNTIL THE CC is applied + even some delay on top of that... this can be told with 100% because the "target of target" gets the rogue after the actuall CC... if it would remove stealth earlier than the CC visually(which would be blind macro/vanish exploitable), there would be target of target instantly, which is not the case here.

Also note that when we fixed this first on Netherwing in 18.3.2017, the cheapshot lost its application sound before it was patched again the very same day. This was also the case at warmane where they hackfixed this after their custom CC-copy pasted delay gone wrong. In their case, the cheapshot never had a sound after that xd.
After few weeks, they completely removed it and left it only for CC type of spells.

tldr - combat proc, aura application(stealth removal, cheap shot aura application) and object delay(visibility of rogue) should be "synced"(not exactly, we will get to it down below) in this case

https://www.youtube.com/watch?v=Ba6Ty72eSH8&feature=youtu.be&t=609
In this video you can see Acrono getting multiple auto attacks in before the combat + cheapshot application + stealth removal aura delay+object delay ... This is just a visual "bug" because the cheapshot was already applied the very first moment GCD golden delay ended... the fact that the cheap shot buff was not VISIBLE on target has nothing to do with the fact that he was already stunned by that time... this is caused by aura application delay not being synced with combat/object delay.

^^ This is very important note note... Aura delay, which we be specific topic of this report has a huge role in this. The stun itself, and therefero the player not being able to move/cast spells has nothing to do with the actual buff/debuff showed up on the PlayerFrame. The Cheapshot invisible stun was applied before the cheapshot aura buff was visible on the frame... Thats extremely important as this completely prevents  the NPC hit/blind exploit from happening.

https://www.youtube.com/watch?v=Ba6Ty72eSH8&feature=youtu.be&t=570
Here you can see Krymu casting gouge from stealth on Acrono. Acrono wasnt able to see him before he was already unable to move/cast spells.

https://www.youtube.com/watch?v=Ba6Ty72eSH8&feature=youtu.be&t=708
Here we can see the following delay order:

0) Acrono casts premeditation instantly - no golden GCD delay
1) Acrono queues cheap shot into golden delay(GCD batch)
2) golden delay
3) golden delay executed, in the very same moment COMBO POINTS are applied to Acrono 
4) delayed stealth batch (takes around 200ms from the end of golden gcd delay)
5) stealth aura buff fades away, cheap shot aura gets visible(aura application batch) + stance bar switches + acrono becomes visible + acrono gets combat --- all 4 synced at the exact same batch

^^ Also note that when 1st golden delay(GCD) finishes, there is ANOTHER ONE that is happening on the stealth buff... It takes around 200ms and when its finished the point 5) happens.


NOTABLE EXAMPLE:

27:32 DESTRAM 4 - the video is no longer available but I have it downloaded in 60 fps quality so you can ask me to upload it...

This is season 4 video and you can see a different sequence of delay pattern:

Screenshot of delay sequences:

Cheap shot gets queued into sapped player:
![image](https://user-images.githubusercontent.com/1942409/47391453-1bc7ca80-d71a-11e8-9e0a-19cfbadd016c.png)

Cheap shot golden delay - gcd batch ends, but stealth is still present, bar stance is still not switched, cheap shot is still not applied:
![image](https://user-images.githubusercontent.com/1942409/47391515-487be200-d71a-11e8-8f96-fb369598d17b.png)

Stealth is removed but cheap shot is still nowhere to be seen:
![image](https://user-images.githubusercontent.com/1942409/47391585-77925380-d71a-11e8-807a-771bf244199a.png)

Autoattacks are displayed while druid is still sapped, cheapshot still nowhere to be seen:
![image](https://user-images.githubusercontent.com/1942409/47391614-8c6ee700-d71a-11e8-8f1c-69b218c2d9cc.png)

You can even see the druid procing the resto talent that procs on crit but he is still under sap:
![image](https://user-images.githubusercontent.com/1942409/47391673-b9bb9500-d71a-11e8-9496-1a2e7070ded5.png)

Cheap + garote + wound  -- aura application delay gets executed
![image](https://user-images.githubusercontent.com/1942409/47391693-c809b100-d71a-11e8-9eca-d50d41da2e76.png)




**NETHERWING VIDEO** ==> https://youtu.be/ZOcVtBZ3CCE?t=55

**Comment by Evolvee** [comment]
**AURA APPLICATION DELAY**
==> also known as "_buff/debuff/aura delay"


https://www.youtube.com/watch?v=jR2US0XPwF0&feature=youtu.be&t=181
Here you can see the raptor getting aggroed before the debuffs are applied.

**NETHERWING VIDEO** --> https://youtu.be/ZOcVtBZ3CCE?t=138
**NETHERWING VIDEO 2** --> https://youtu.be/9jaxWhjnbsE?t=11
**NETHERWING VIDEO 3** --> https://youtu.be/EI3xJepTr30?t=13
**NETHERWING VIDEO 4** --> https://www.youtube.com/watch?v=CyoMz8T_Ol0
**NETHERWING VIDEO 5**  --> https://www.youtube.com/watch?v=W3l6OEo9Wzg

Its self explanatory. Basically all auras, debuffs, buffs, effects are applied and removed in a batch. How the multiple batches are synchronized with each other determines the outcome of this and the final effect that you can see in those videos....

Tuska 4 ft. Boozt 10:30 https://youtu.be/5HnthGAKrn8?t=628 You can see the golden GCD delay gets executed and silence appears in ANOTHER batch(aura application batch) after that... not instantly.

This mechanic is also responsible for why you can see in retail videos that ppl are still in cc animation when they trinket for some time before the animation fades out into normal character state without any CC. It looks like they are running in stun/sap because of the aura application and object delay. 
E.G. https://youtu.be/Ba6Ty72eSH8?t=529 you can see that after he gets out of sap, he is still moving with his head down, not narrow... This is not the best footage to show this mechanic but Im pretty sure you can find PLENTY out there on the internet... just watch some arenas or rogue duels.

https://youtu.be/_sX14VJj5X4?t=150 You can see he isnt able to cast anything while the Intimidating Shout still didnt appear in his debuff bar and then you can see him being able to cast while the intimidating should is still visible in the debuff bar.


----------------------
**MIND FLAY**
=> Also known as _channeled beam delay_

https://youtu.be/jR2US0XPwF0?t=218
https://youtu.be/jR2US0XPwF0?t=73
https://youtu.be/veOfNNoBdOk?t=823 (season 4 footage)

**NETHERWING VIDEO** --> https://youtu.be/NeGqoP4zHb0?t=19

So basically, as you can see... sometimes when you queue mind flay, the mind flay is not always executed instantly... Sometimes there is a delay between mind flay execution(channeling) and the actual beam + debuffs tied to the beam(such as shadow weaving, mind flay slow component, mind flay periodic dmg components, misery etc.)

The correct order(in case this mechanic happens) is:
1) golden GCD delay
2) mind flay executed, channeling bar appears counting down
3) delay
4) beam + mind flay debuff(dmg + slow component) appears in the same batch
5) delay
6) misery appears delayed in this next batch

As you can see on the videos above, the mind flay BEAM batch is "synced" with aura application delay(described above). When you watch the Menismyforte 2 clip above, Mind Flay beam + Mind Flay debuff appears at the exact same moment when Vampiric Touch fades off...
Why "synced"? If you watch it VERY closely, you will notice that the aura application delay happened FEW miliseconds prior to the beam visibility.
Look:

1) Mind flay still not applied, vampiric touch is about to fade off.
![image](https://user-images.githubusercontent.com/1942409/47878894-9d58e000-de1f-11e8-9a67-7a890288b412.png)

2) Mind flay debuff visible, beam still not. Vampiric Touch faded off successfully.
![image](https://user-images.githubusercontent.com/1942409/47878971-d09b6f00-de1f-11e8-82a8-e74312a232d9.png)

3) Few miliseconds after, beam becomes visible.

![image](https://user-images.githubusercontent.com/1942409/47878997-e1e47b80-de1f-11e8-8cd5-b2bfe46f25c8.png)

4) When Mind flay is almost in the half of its duration, Misery gets refreshed:
![image](https://user-images.githubusercontent.com/1942409/47879031-f163c480-de1f-11e8-84b7-bad49c0a1d27.png)


**! IMPORTANT!**
Note that when you first see the Mind Flay on the target, the Mind Flay timer is already down by 1/5 of its "duration":
![image](https://user-images.githubusercontent.com/1942409/47879132-25d78080-de20-11e8-9b52-0c45de9cd3dd.png)
 ^^ This is the very first frame after Mind Flay debuff is visible. You can see its visual timer on the victim frame is not 100%, but rather around 80% full. This is because of aura application delay. If there was no aura application delay, you would see the first frame of mind flay as 100% full, but because its delayed, we cant see it until the appropriate batch gets executed.
The same exact effect happens at the end of the mind flay cast where you see 0 sec duration on the debuff, but the Mind Flay is still slowing down and able to reproduce one more last 3rd tick.


Dont worry, its not that easy. During our research, we figured out that if you queued Mind Flay into already happening cast(we will get into spell queue mechanic later) due to golden GCD delay, the beam would ALWAYS be instant, basically skipping the "channeled delay" and going straight to the actual beam(which contains all spells attributies)... the channeling itself is not part of the mind flay per se, the beam is. Same will most likely(I didnt research it) be applied to Drain Life and Drain Mana.

^^ This above was never attemted to be fixed on NW because we didnt know whats causing it and how it works in the first place. But it should be noted, that its there.

![image](https://user-images.githubusercontent.com/1942409/47878151-e314a900-de1d-11e8-8291-3995e65a0ddf.png)

_Recommended videos to watch for research: Menismyforte 3, Menismyforte S4 2200 shadowpriest tbc(S3 vid), Vengeful Gladiator Tuska and Boozt - World 1 SpriestRogue S3  2700  TBC [HD]_

**! IMPORTANT/INTERESTING !**

https://youtu.be/veOfNNoBdOk?t=878
![image](https://user-images.githubusercontent.com/1942409/47881847-bf566080-de27-11e8-99a2-d2d0da2be8fa.png)

In this video you can see Mind Flay being queued into Mind Blast. When GCD batch procs, you can clearly see mind flay beam having no delay and being synced with both: resist AND damage text. The beam is clearly visible before mind flay debuff in this case.

Mind Flay is a binary spell, that means it either resists fully or not at all - its ticks cannot be resisted separately nor partially. That is very weird because as you clearly can see the first dmg was 641 dmg while 2 subsequent ticks were for exactly the same = 456. So mind flay is 456 and mind blast was partially resisted to value of 641. What is the resist text for then?
![image](https://user-images.githubusercontent.com/1942409/47881930-093f4680-de28-11e8-8b90-de8e2ec9afb3.png)

When aura application delay procs mind flay starts visually from around 80% timer and the shadow weaving says 2 stacks, which means mind blast + mind flay(what is the resist from then??). Misery procs correctly in delay around 60% of the mind flay duration.

-----------------

**CHARGE / INTERCEPT vs Aura application delay**

Many ppl, including corecraft (https://youtu.be/XeCiG5eJyfE?t=44) though that intercept/charge stun should be delayed. That is however not correct.

Intercept/Charge stun and its interrupt effect(same as all other interrupt effects in the game btw) should be instant:
https://youtu.be/w0CrmZygCuY?t=272

https://www.youtube.com/watch?v=3z2nV3pRDS0&feature=youtu.be&t=32

As you can see in the video, the cast is INSTANTLY interrupted while the stun is applied in delay. BUT! You need to be smart and remember what we said about aura application delay... THE STUN/INTERRUPT effect is launched instantly, synced with golden GCD delay, while the aura representing that is delayed only visually. That makes you think the stun should be delayed because you see the intercept stun in debuff showed delayed, but thats only visual thing. Server-side its instant and you wont be able to cast anything before you are visually affected by the stun either.

---------------------

**CONE OF COLD**

Noone 1 6:45 https://youtu.be/-J55f0puu9A?t=404 ; https://youtu.be/-J55f0puu9A?t=251 

Damage applied instantly (all damaging are INSTANT), cone of cold debuff in next aura application batch, winters chill double delayed(because its a proc) into 2nd aura application batch.
Its worth to mention that all procs that are derived from main spells are ALWAYS delayed into next aura application batch after the main spells aura application batch --> e.g. shadow word pain + shadow weaving, misery in delay...

Noone 1 16:41 https://youtu.be/-J55f0puu9A?t=1001

Explanation of this clip in **NETHERWING VIDEO** => https://youtu.be/CyoMz8T_Ol0?t=97

Its pretty well explained there...
The delay order is as this follows:
1) golden GCD delay executed, spell launched, damage processed instantly(gets postponed into delayed HP update batch later on--> we will get to this later)
2)delay
3) aura application batch procs --> cone of cold + frostbite applied in the same batch synced together
4) delayed
5) next aura application batch executed --> winters chill is applied

You can also see that mana is consumed in delay bound on the combat batch, but we will get to that later. This is STRICTLY about visual part of the spell.

**Comment by Evolvee** [comment]
**IMPROVED COUNTERSPELL**

**NETHERWING VIDEO** => https://youtu.be/CyoMz8T_Ol0?t=12 ... its pretty self-explanatory in this video.

Improved Counterspell should have SEPARATE chances to:
1) be resisted as a whole spell;
and if its not resisted
2) have chance for the "improved" part of the counterspell to be resisted since it should be treated as its "PROC"

Its basically a partial resist sui generis(im a law student btw xd).
On retail, you could imp. counterspell someone, lock the spell school for 8 seconds but the silence part of counterspells(which gets applied in delay) could be resisted - shown in Noone video.
Noone 1:
https://youtu.be/-J55f0puu9A?t=404
https://youtu.be/-J55f0puu9A?t=905
Noone 2:
https://youtu.be/K4WR0JW5jWM?t=907 (on focused druid)
Noone 3:
https://youtu.be/IAoe3akMZUE?t=128


In all of those clips above you can see that the interrupt effect is instantly launched after the golden GCD batch gets executed. Yes, same as charge/intercept. And the improved part of the spell(flat silence) should appear delayed in next aura application batch.


---------------------------

**CLOAK OF SHADOWS**

**NETHERWING VIDEO** => https://www.youtube.com/watch?v=84D_mhQRsgE (this entire video is dedicated to cloak), should be pretty self explanatory..

Basically cloak should protect you from magic effects before you get the cloak itself through aura application batch. This is because the IMMUNITY and the BUFF part of the spell are separated. The immunity is there way before the actual buff, making you to be able to react to spells.

https://youtu.be/fT_aZfg9AP0?t=408
The video shows getting the "IMMUNE"(same as fear ward... coincidence? i dont think so) when golden GCD delay gets executed and before the cloak is even applied due to aura application delay.

I believe my Netherwing video explains this in detail already, so no need to do some essay writing in here.

---------------------

**IMPROVED BLIZZARD**

Basically, blizzard should deal damage first and THEN apply the slowing component in aura application batch.
https://youtu.be/E8hQG5CnG5s?t=13

**NETHERWING VIDEO** => https://youtu.be/W3l6OEo9Wzg?t=13

Self explanatory, lets move on.


-------------------

**MIND BLAST - showcase of multiple batches being and not being synced with each other**

**NETHERWING VIDEO** => https://youtu.be/W3l6OEo9Wzg?t=71

Menismyforte 2: Devour - 8:50 , 16:59

https://youtu.be/_sX14VJj5X4?t=527 (best)
https://youtu.be/_sX14VJj5X4?t=1017


Shortly summarized: Combat proc should always be synchronized with delayed mana consumption batch (will be explained later). Delayed HP batch should be synchronized with first aura application batch that comes after the mind blast application batch.

Just watch the Netherwing video above.

-----------------

**Shadow Word: Pain**

A brilliant example of the actual delay and delayed aura application!

**NETHERWING VIDEO** => https://youtu.be/9jaxWhjnbsE?t=11

Its self explanatory... Pain always procs shadow weaving at the same batch while Misery gets always delayed into another aura application batch... its also semi-explained above when I was talking about how procs are handled.

Menismyforte 2: Devour - 9:50 , 12:37 , 16:40

Pain + Shadow weaving is instant btw, same as any other spell. Its just aura application delay that makes it look like delayed.

-----------------

**SHADOW WORD: DEATH**

**NETHERWING VIDEO** => https://youtu.be/EI3xJepTr30?t=11

retail: https://youtu.be/ch3aXvhmADc?t=203


Explained in the netherwing video.... its basically instant dmg cast, same as mind blast or any other spell... the dmg is instant but it looks like delayed because of delayed HP updates.

The way SWD into POLY works is that you queue SWD at around <400ms prior to poly cast launch. If you manage to queue(golden gcd delay) that SWD inside, u have 100% chance to break poly because:
1) u queue SWD
2) poly cast ends
3) batch procs --> both swd and poly is launched at the same time - swd deals dmg and poly is applied in next batch
4) poly is applied and then backlash SWD dmg procs, interrupting applied polymorph --> you can notice that on retail, the delay between being affected by poly and removing the poly by SWD backlash dmg is always the same => its always the same batch that is responsible for it --> so it doesnt matter when you launch SWD, all what matters is, if you did it <400 ms prior to end of the cast. If you did, you are always guaranteed to have it processed in the same priority as poly. **THIS HOWERVER COUNTS WITH GOLDEN GCD DELAY BEING 400 MS WHICH WE WILL DEBUNK LATER!** --> its rly not that case in S4+ videos.

There is a common public misinterpretation of this mechanic in spell delay system that is being spread by ppl who dont understand what spell delay even is. Most ppl believe that this works only because SWD damage is delayed, essentially not making pusback to polymorph cast at the end of the cast, making the SWD backlash dmg successfully interrupt the polymorph. But thats just not how its shown in the retail video above, making this claim completely irrelevant and disproved.

Instant spells were never delayed on retail unless you queued them(except golden GCD delay... without this delay the system will go to shit). Thats why Mind Blast + Swd combo always deals both damages at the exact same time and not divided into 2 batches.

--------------------

**Inner focus + Devouring Plague**

https://youtu.be/_sX14VJj5X4?t=897
https://youtu.be/_sX14VJj5X4?t=166
--> this is just an offtopic
As you can see in the video, when you use these 2 together with Inner Focus being 1st, the Inner Focus will NEVER appear in your buffs because of aura application delay... The inner focus is consumed faster than the aura application batch can be processed. This is also another confirmation that the general golden GCD batch, the global delay batch and the aura application batch IS NOT SYNCED and PROCESSED together.


---------------

**DRINKING AFTER CASTING HEALING SPELLS**

I think we dont have to say much about this, its pretty well known.
The way how it works is that you QUEUE the drinking at the end of healing spell cast. You can queue it inside another queued spell as well and it will go off.
For example:
You are casting Holy Light, 300ms prior to end, you queue mending into golden GCD delay and right after u queue drink. Both holy light and mending will go off and u will be drinking.
BUT
If you do this with instant, you wont be able to do it since there is nothing to queue into it. U cast mending, and because of combat batch(which we will get into later) you get instant combat, you wont be able to drink.
If you however queue drink at the end of a cast, it will work....
Also note that this aparently works only for HEALING spells, not spells like frostbolt(which triggers the combat immediately upon launching ---> not always, but we will get to this later)

If you need more info about this, there is a plenty of reports on old forum and old BT. I cant link it ^^

Notable examples: Hydra and Tehseus(pala+sham) videos.

**Netherwing video (the showcased mechanic is not correct, its outdated implementation):** https://youtu.be/sfx6EXU0U68?t=102

http://www.arenajunkies.com/topic/31577-drinking-bug/
http://www.arenajunkies.com/topic/19433-healers-do-you-combat-drink-exploit/#entry154793
http://www.arenajunkies.com/topic/19433-healers-do-you-combat-drink-exploit/page-2#entry155072

Its possible that offensive spells applied combat event instantly server-side but server sent them to users(client) in batches.

**Comment by Evolvee** [comment]
**SPELL QUEUE SYSTEM**
=> also known as golden delay, golden GCD delay, GCD batch (or how ppl with no clue call it --> "initial spell delay")

**NETHERWING VIDEO** => https://youtu.be/FXFjvsD7VI4


https://www.reddit.com/r/wow/comments/624y53/attn_custom_lag_tolerance_was_automatically_set/dfk1ev1/
https://wowhats.wordpress.com/2010/03/07/lag-wow-and-you/
http://ropetown.com/phpbb/viewtopic.php?t=2582
https://web.archive.org/web/20080219100205/http://forums.worldofwarcraft.com:80/thread.html?topicId=879058320&sid=1&pageNo=11
https://web.archive.org/web/20080408033047/http://elitistjerks.com:80/f47/t15393-global_cooldown/
https://web.archive.org/web/20071023182508/http://elitistjerks.com:80/f15/t17332-2_3_changes_10_19_a/ (post \ 66)
https://web.archive.org/web/20080304235309/http://elitistjerks.com:80/f31/t16979-mage_tc_after_2_3_a/p3/
https://www.engadget.com/2007/11/05/dont-mash-in-2-3/


Basically on retail, there was a separate batch that handled how spells are being launched(GCD was literally on a batch). On most retail videos, you can clearly see that when ppl click on their spells, they are being QUEUED. That means that the action button where the spell is located becomes "golden"... thefore "golden delay".. After the golden GCD batch ends, the button becomes gray again and the spell is launched.

Example:
GOLDEN DELAY:
![image](https://user-images.githubusercontent.com/1942409/47934242-1ebf7980-ded7-11e8-8e61-55abc4c8c339.png)
NO DELAY:
![image](https://user-images.githubusercontent.com/1942409/47934255-297a0e80-ded7-11e8-921b-c3b0000222c2.png)

What golden delay does is making sure that ALL spells on the entire server are launched at the EXACTLY same time... at the end of this "golden gcd batch".

What it also does is that it ENABLES you to queue spells. Literally. If you cast a spell(lets say Holy Light) and then click on some other spell(lets say flash heal) when u are <400ms to the end of the first cast, it will BYPASS the initial delay that you get due to golden GCD delay and goes straight to cast of that 2nd spell(flash light) without any additional delay.

Which means: If you queue spells properly, you are gonna be affected by this "golden gcd delay" only once - at the very first cast. After that you are bypassing this delay by queueing into spells that are already in progress of casting.

This also increases the DPS of casters since its allows them to cast their spells with no micro-downtime between casts!

I believe this is also affecting how hunter PVE macros work. This enables you to queue Steady Shots one after another with 100% certainty. 

If you watch the showcased Hydra video closely, you can see he is spaming the mana burns over and over again and when he is finally at around 0.5 sec until end of the cast. The mana burn GETS THROUGH, successfully bypassing the GCD check and allowing him to bypass the GCD. Once the first mana burn spell finishes, the 2nd one INSTANTLY starts again with 0 downtime caused by latency. 


**!!!BUT!!!**

It seems that this mechanic was heavily observable only on early TBC patches. Lets say Season 1-3. In season 4 videos, this delay is almost non existent, happening very rarely and if it happens, then its usually around max. 200ms. Certainly not 400ms.

Question is how can this system work then... did retail shorten the initial golden delay(which was on Netherwing beta full fledged 400ms before it was removed) or did they change the system drastically to work in completely different way? Probably the 2nd option which would also explain all the other changes made to GCD and shot queueing around patch 2.3.

What is known is that around patch 2.3, blizzard changed the queuing mechanisms that also allowed hunters to do their famous 3:2 rotation etc. This might very well be tied to this mechanic rework.

Patch 2.3 discussion: https://web.archive.org/web/20071028151030/http://altitis.blogspot.com/2007/10/stopcasting-quartz-and-patch-23.html

Currently the GCD delay on NW is fully disabled, there is 0 ms on GCD batch but its allowed for the client to queue as well. So you get no delay at the start but you are allowed to queue at the end of the cast. Pretty much win-win scenario. What impact does this have on spell mechanics is unknown yet. I think it might interefere in how double fear ward / cloak / vanish works.

--------------



**DELAYED HEALTH UPDATES**

**NETHERWING VIDEO** => https://youtu.be/r7CSJqQOypc?t=132

https://web.archive.org/web/20080625061700/http://elitistjerks.com/f32/t26517-instant_health_updates/
https://us.battle.net/forums/en/wow/topic/12878868036
http://www.lurkerlounge.com/forums/thread-1873.html
https://wow.curseforge.com/projects/project-6450

Basically what this does is that ALL heal and damage actions are happening in batches rather than immediately. The batch that processes this is called "delayed hp update". 

Note that this DOES NOT affect UNIT_HEALTH event combat log. Combat log still logs all actions immediately, On retail, ppl could install the addon listed above to see their CURRENT HP(which took the data from combat log) before it was processed in a batch and sent to a client. This resulted in ppl seeing negative HP values sometimes. This is because you could literally go under 0 health and still live if u received a heal in the very same batch that would put you at 1+ HP at the end of the batch where all dmg and healing actions are being processed.

This is also very obvious on retail pve videos where you can see that tanks HP is not going down immediately but rather in "CHUNKS". This is because of delayed HP update batch.



---------------


**DELAYED MANA CONSUMPTION**

**NETHERWING VIDEO** => https://youtu.be/0cRn47JPXz4?t=71
**NETHERWING VIDEO 2** => https://youtu.be/2xR8fYmFqUc

(note that in the first video, the winters chill hits the same aura application batch as Cone of Cold, which I have no idea how could that happen --> if you go back to aura application or CoC topic, you will know why)

Mana consumption delay should be synced with combat batch. Mana and Combat events should happen at the same time. 

The 2nd video showcases a phenomenon that I dont know why sometimes on retail happened yet.
The 1st video showcases that mana consumption should be subject to spell delay as well.
It seems that:

1) Casted flying spells should proc instantly combat, so they drain the mana instantly.(except for the case in 2nd video)

2) Instant flying spell should proc combat in batches and when the combat procs, at the exact same time mana drain procs, so its linked together at the end of the batch.

Now, lets stop here for a second and brainstorm about what we already know. We know that combat duration is tied to energy ticks. We know that mana consumption is tied to combat proc. 
What if mana consumption (and delayed health updates) are tied to energy ticks as well but rather in 5x 400ms (1 energy tick of 2 seconds). That would make combat timer unlikely to be 4.5-6.5 sec as we currently have it. What is the evidence behind that after all? I only remember Schaka talking about it and giving these direct numbers but they were never rly backed up with any proof... On warmane there is a bugreport where they pretty much try to negate this theory. And they _might_ be right:
https://www.warmane.com/bugtracker/report/46231


On multiple retail videos, especially Noone, you can observe this:
1) noone queues cone of cold
2) cone of cold gets executed
3) cone of cold deals damage
4) delay
5) after aura application delay, cone of cold applies CoC slow debuff synced with mana consumption
6) delay
7) cone of cold procs Winters Chill in aura application delay

Now lets stop again. At point 5. If mana consumption is synced with aura application delay. It means that combat is synced with aura application delay as well, therefore all of these 3 batches are synced together? Loominatey confirmed?

--------------------------


**DELAYED COMBAT**

**NETHERWING VIDEO** => https://youtu.be/r7CSJqQOypc
**NETHERWING VIDEO 2** => https://www.youtube.com/watch?v=2xR8fYmFqUc

https://youtu.be/K4WR0JW5jWM?t=771
procs midway

https://youtu.be/K4WR0JW5jWM?t=688
procs almost instantly

https://youtu.be/-J55f0puu9A?t=1168
instantly on frostbolt launch

https://youtu.be/GKOzjFoLn_s?t=34
instantly after FB is launched, he gains combat

https://www.youtube.com/watch?v=RMV-fuwQiz0&feature=youtu.be&t=20
Well... the frostbolt didnt trigger combat instantly here :(

https://youtu.be/M_P5NocW_Ew?t=22
ice lance proced combat instantly here

https://youtu.be/kfgqR_cTBG8?t=32
frostbolt procs midair

https://youtu.be/kfgqR_cTBG8?t=45
frostbolt procs midair

https://youtu.be/kfgqR_cTBG8?t=94
frostbolt procs instantly

https://youtu.be/EAJUvk9WNKY?t=13
procs instantly

https://youtu.be/FLEhudSh51U?t=24
procs instantly

https://youtu.be/FLEhudSh51U?t=45
procs instantly

https://youtu.be/FLEhudSh51U?t=103
procs instantly

https://youtu.be/FLEhudSh51U?t=111
procs instantly(maybe 50ms delay)

https://youtu.be/FLEhudSh51U?t=149
procs instantly

https://youtu.be/FLEhudSh51U?t=186
procs instantly 100%

https://youtu.be/FLEhudSh51U?t=341
procs in delay

https://youtu.be/FLEhudSh51U?t=443
pom + fireball ==> procs instantly

https://youtu.be/FLEhudSh51U?t=462
procs midair

https://youtu.be/FLEhudSh51U?t=478
procs almost instantly, projectile is already in air when combat procs

--------------


**DOUBLE FEAR WARD**

**NETHERWING VIDEO** => https://youtu.be/NeGqoP4zHb0?t=108

![image](https://user-images.githubusercontent.com/1942409/47952603-765ef300-df72-11e8-9c8d-e6ff50e1bb74.png)

^^This could never happen on retail.

ccshiro/cc-buglist#1479
https://www.youtube.com/watch?v=KoQDA9mRDts#t=16m51s

The fear ward basically provides immunity before the aura application batch procs(similar to what cloak/ice block does). That means it "IMMUNES" incoming fear effects even before you get the fear ward itself. That way you can get 2 fear wards. You need to cast the fear ward literally INSIDE the batch that handles the fear effect. If you do it batch closer or batch later, ur fucked.

This has to work both against:
1) instant fears (physich scream)
and
2) casted fear (Fear, Howl of Terror)

AND

a) direct fears (Fear)
and
b) aoe fears (howl of terror)


**Note!**
When you manage to achieve this mechanic, the fear ward **shouldt** say "IMMUNE"(when the 1st fear aura application delay is executed) in scrolling combat text. It should say fucking nothing.. nada.


-------------------

**DELAYED VISIBILITY SYSTEM / OBJECT DELAY**

This was already partially mentioned in Vanish topic.
In Black Morass sniffs, you can see that objects are appearing in around 400ms intervals. Sounds familiar?

**NETHERWING VIDEO** = >https://youtu.be/ZOcVtBZ3CCE?t=59


---------------


**HELLFIRE VS CC**
_==> Potentially working with Life Tap / Blood Rage as well_

This never-seen-on-any-private-server-in-history mechanic is pretty cool.

retail footage(WOTLK): https://www.youtube.com/watch?v=MkRUlAHMFsU&feature=youtu.be&t=620
https://github.com/ccshiro/cc-buglist/issues/1560
http://www.gamefaqs.com/boards/534914-world-of-warcraft/54935109
http://www.wowhead.com/spell=1949/hellfire#comments
http://www.skill-capped.com/forums/showthread.php?19670-Question-about-Hellfire...
http://www.arenajunkies.com/topic/88945-hellfire-sheeps/
http://www.ownedcore.com/forums/world-of-warcraft/world-of-warcraft-exploits/145833-break-polymorph-warrior.html
http://www.ownedcore.com/forums/world-of-warcraft/world-of-warcraft-guides/146099-warriors-breaking-cc-pvp.html


**NETHERWING VIDEO** => https://youtu.be/5XBimiJ1BM0?t=50

It most likely works like this: Polymorph hits - u take dmg in between spell hit and effect application - Polymorph effect applies and cancels instantly.
So how this probably works?:
1)	Polymorph cast
2)	Hellfire cast 
3)	Polymorph hits(sheep sound)
4)	Hellfire dmg procs
5)	Polymorph cancels itself before the polymorph effect applies – OR it happens in the exact same batch as HP update procs and due to aura application delay being postponed (into step 6.)
6)	Now normaly there would be another sheep sound and smoke cloud effect + polymorph visual and debuff on target, but this phase seems to be entirely skipped due to 5)

-----------

**BLINK TELEPORT / VISUAL DELAY**

https://youtu.be/1OaMCfCDMt4?t=79

Pretty publicly-known "mechanic" where the blink spark appears before the player is teleported. Tied most likely to object delay.

-----------

**SPELL LOCK**

--> Works exactly as **Improved Counterspell**
https://youtu.be/1OaMCfCDMt4?t=83

**Comment by Evolvee** [comment]
### **Notable mentions / retarded exceptions and random clips that support claims writen above(or not):**

https://youtu.be/4dSNU--fbDQ?t=98
look at when wlock casts fear
the fear is interrupted before kidney shot is visible
its interrupted in the exact same moment as when the kidney shot on rogues action cast bar loses the "golden delay"



https://youtu.be/4dSNU--fbDQ?t=219
heal is interrupted instantly when blind golden delay fades off
heal is interrupted even before the blind is visible
now... this could also be the delayed aura visibility but again.... its interrupted at the same moment as when the golden delay is off



----------------

Menismyforte 2: Devour https://youtu.be/veOfNNoBdOk

16:30+... in those duels all his spells were instant, not a single "golden delay"

16:46 the power word:shield was in his buffs even before the visual was applied(buble) and before cooldown started... its instant on click...

![image](https://user-images.githubusercontent.com/1942409/47953059-3b5fbe00-df78-11e8-8b23-ff1381c8dad1.png)

-----------

channeling spells shouldnt be always delayed

-currently there is ALWAYS that beam delay
-on retail there was only sometimes:

instant drain life = https://youtu.be/7k30ZBJ23MQ?t=119

-------------

https://www.youtube.com/watch?v=iiFbrMUegsI
https://youtu.be/JfQ-H6dos3w?t=102
https://youtu.be/E8hQG5CnG5s?t=10

on these videos u can see that hp update is synced with dmg dealt by blizzard(except the chinese one) and that blizzard has like 1 sec delay
also on the chinese one u can see that the blizzard dmg is represented by some projectile falling on rogues head(for some reason its present even before the dmg lul xd)
also the imp blizzard debuff on NW is not periodic... due to visual delay the blizzard on retail vids is periodically refreshed
but on NW the 1st imp blizzard is almost instantly replaced by 2nd slow tick of imp blizzard and THEN its periodic...
also you can see that mana consumption is bound to channeling bar(same as on mind flay) but on NW its bound to damage
alsoo the chinese blizzard seems retarded af.. it reveleas rogue when rog is already behind pillar range , maybe cuz of icy veins(haste buff)?

---------

Trinkets/Item use should probably be excluded from spell/golden delay:

https://youtu.be/RbeqSGY0tM8?t=13
-he trinkets INSTANTLY as he pressed the trinket, there is no "golden delay"


-------------


https://youtu.be/gE53OciwosA?t=7
when he finishes the cast, there is the end cast animation where he rises his hands up
that doesnt happen on NW


------------

https://youtu.be/E8hQG5CnG5s?t=10

look at this video at beggining

the frostbolt hits into poly, HP UPDATE IS INSTANTLY SYNCED with damage received and THEN winters chill procs...

on mindblast footage from menismyforte 2 : mindblast casted ,mana consumption, dmg dealt, HP update DELAYED SYNCED WITH SHADOW WEAVING APPEARENCE... (basically hp update and sh weaving synced.... on frostbolt video winters chill and hp update NOT synced)
also there is no frostbolt debuff xd


----------

Spell Queue should probably count with proced auras before next spell starts?

ccshiro/cc-buglist#2521
exactly the same issue
when u queue holy light, the 2nd holy light is not reduced in cast when u have this talent
ccshiro/cc-buglist#1890
![image](https://user-images.githubusercontent.com/1942409/47953137-7a424380-df79-11e8-9cf1-4e801efe2059.png)

-------------

menismyforte2: https://youtu.be/veOfNNoBdOk 7:37

mana is consumed when mass is finished and says "immune"
"immune" message appears instantly upon mass dispel hits the ground
on netherwing mana consumption is delayed as fuck and dispel is probably delayed as well

------------------


Queueing mind flay(into golden delay after spell cast of another spell) should bypass its initial "beam delay"
-on retail vids, the beam is instant upon 1st spell is finished
-on nw there is a delay between these 2



--------------


![image](https://user-images.githubusercontent.com/1942409/47953152-cdb49180-df79-11e8-93b9-fcfb0773069d.png)

![image](https://user-images.githubusercontent.com/1942409/47953156-d1e0af00-df79-11e8-8e08-b65471d4f021.png)


----------


Delayed HP updates vs NPC death status

-when u land lethal hit on NPC, the npc stops and doesnt do anything until the HP update ticks and it dies...
-it should attack/follow/whatever until it dies, not when the hp is below 0 (like with players) probably --> looks scuffed af otherwise


----------------------

delayed hp update vs ressurection trigger

idk if this still happens on NW but it did in 2017:
check: https://www.youtube.com/watch?v=8z5oiqMd8o0&feature=youtu.be
if u receive lethal and then it heals u , u will get the release spirit shit and it cancels itself instantly


-----------------
**This report was written in 2017, idk if its still valid!!;**

Stealth vs Action bar spells

-When you click steath and after that "1" button where u have some dmg ability on normal bar and sap ability on stealth bar, it casts the dmg ability even before u enter stealth....

-i think that is not correct, currently u have to wait loooooong as fuck before u enter stealth --> cast sap

-logically if u cast 1 ability and AFTER 2nd ability... the 1 should finish first

-in combat log the stealth is waaaay before dmg ability(lets say sinister strike)

needs research

29:30
https://vid.me/ZqBm

720p 60 fps video btw, there is a lot of re-stealths

---------------

**Comment by Evolvee** [comment]
reserved

**Comment by Evolvee** [comment]
reserved

**Comment by Evolvee** [comment]
reserved

**Comment by Evolvee** [comment]
reserved

**Comment by Evolvee** [comment]
reserved

**Comment by femdead** [comment]
don't you think fixing bugged class mechanics is more productive than this

**Comment by Palabola** [comment]
> 
> 
> Spell delay / batching was not a design decision, it was a limitation put in place by the hardware and infrastructure blizz had back in the day. The exceptions you have observed in your videos look just a combination of timing randomness between different batches and latencies :D Not an intended and desired system but the opposite.
> So why has it been added? Just because the monkeys on wowservers reddit like to use buzzwords they heard once doesnt mean it is a good idea :D

I made the same conclusion about what we seen on the videos. Some exceptions are exist, like Shadow Word: Death + Poly and the Grounding Totem.

**Comment by netherspite** [comment]
> When aura application delay procs mind flay starts visually from around 80% timer and the shadow weaving says 2 stacks, which means mind blast + mind flay(what is the resist from then??). Misery procs correctly in delay around 60% of the mind flay duration.

The resist text could be for the Blackout proc.

**Comment by Evolvee** [comment]
True lol, I totally forgot about Blackout probably.
